schnorr_fun/src/keypair.rs

```diff
@@ -28,8 +28,8 @@ impl KeyPair {
     }
 
     /// Returns a reference to the public key.
-    pub fn public_key(&self) -> &XOnly {
-        &self.pk
+    pub fn public_key(&self) -> XOnly {
```

Comment:
Any rationale on why you go this way (by value instead ref?)

Reply:
`XOnly` is Copy so I *think* it's conventional to not return references to `Copy` things. On top of that There's no reason to leak the implementation that `keyPair` stores an `XOnly` (it could store a `Point<EvenY>` instead). 

(that comment needs to change though).